core/dbt/task/generate.py

```diff
@@ -227,10 +228,14 @@ def run(self) -> CatalogArtifact:
         if self.manifest is None:
             raise DbtInternalError("self.manifest was None in run!")
 
-        adapter = get_adapter(self.config)
-        with adapter.connection_named("generate_catalog"):
-            fire_event(BuildingCatalog())
-            catalog_table, exceptions = adapter.get_catalog(self.manifest)
+        catalog_table: agate.Table = agate.Table([])
+        exceptions: List[Exception] = []
+
+        if not self.args.empty_catalog:
+            adapter = get_adapter(self.config)
+            with adapter.connection_named("generate_catalog"):
+                fire_event(BuildingCatalog())
+                catalog_table, exceptions = adapter.get_catalog(self.manifest)
```

Comment:
Just for readability sake, can we re-write this such that it's a little more clear what code is being executed on either side of the conditional and do so without re-assigning variables?  Maybe something like:

```
if self.args.empty_catalog:
    catalog_table: agate.Table = agate.Table([])
    exceptions: List[Exception] = []
else:
    adapter = get_adapter(self.config)
    with adapter.connection_named("generate_catalog"):
        fire_event(BuildingCatalog())
        catalog_table, exceptions = adapter.get_catalog(self.manifest)
```

Reply:
Thanks for flagging, fixed!